### PR TITLE
Feature/form

### PR DESCRIPTION
**Pull request by KyleWong2510**
## Pull request checklist
Please check if your PR fulfills the following requirements:
- [ ] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type
 
 
Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe): 

## What is the current behavior?

-  No Redux or form functionality

## What is the new behavior?

- Implemented Redux and thunk
- Wrote all actions
- Reducer boilerplate
- Add form functionality
- Fetch data on submit
- Update store with form data

## Does this introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: src/App/App.js [hunk 1]
```diff
@@ -2,30 +2,48 @@ import React from 'react';
...
-    </main>
+        </div>
+      </main>
     )
   }
   
   return (
     <main className='App'>
-    <Form />
+      <Switch>
```
Comment (Billwilke42) [inline src/App/App.js:31]: Nice job getting router implemented

File: src/Form/Form.js [hunk 2]
```diff
@@ -1,74 +1,90 @@
...
+    this.setState({ [name]: value })
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault()
+    this.props.setPlayerName(this.state.name)
+    this.getAllQuestions()
+  }
+
+  getAllQuestions = () => {
```
Comment (Billwilke42) [inline src/Form/Form.js:37]: So glad we got this working

File: src/Form/Form.js [hunk 3]
```diff
@@ -1,74 +1,90 @@
...
-          <option>
-            Science & Nature
-          </option>
-          <option>
-            Books
-          </option>
-        </select>
-    )
+    let inputs = []
+    for(let i = 1; i < 7; i++) {
```
Comment (Billwilke42) [inline src/Form/Form.js:48]: This was a doozy

File: src/reducers/setQuestions.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      return action.message
+    default:
+      return state
+  }
+}
+
+export const setQuestions = (state = [], action) => {
+  switch (action.type) {
+    case 'SET_QUESTIONS':
+      return [...state, action.questions]
```
Comment (lilianaweimer) [inline src/reducers/setQuestions.js:22]: This is great, I'm so glad that this works!

File: src/Form/Form.js [hunk 5]
```diff
@@ -80,4 +96,17 @@ class Form extends React.Component {
...
-export default Form
\ No newline at end of file
+// const mapDispatchToProps = dispatch => {
+//   return {
+//     getQuestions: (cat, diff) => dispatch(getQuestions(cat, diff)),
+//     setPlayerName: () => dispatch(setPlayerName())
+//   }
+// }
+
+const mapDispatchToProps = dispatch => (
```
Comment (Billwilke42) [inline src/Form/Form.js:106]: Maybe we should ask one of the instructors why we need bindActionCreators here.

File: src/actions/index.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
+export const setQuestions = questions => ({
```
Comment (Billwilke42) [inline src/actions/index.js:1]: Nice job getting ahead on the actions file

File: src/reducers/index.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import { setCurrentQuestion } from './setCurrentQuestion' 
+import { setScore } from './setScore'  
+import { setLives } from './setLives'  
+import { setPlayerName } from './setPlayerName'  
+import { setLeaderBoard } from './setLeaderBoard'
+import { isPlaying } from './isPlaying'
+import { isLoading } from './isLoading' 
+import { hasErrored } from './hasErrored'
+
+export const rootReducer = combineReducers({
```
Comment (Billwilke42) [inline src/reducers/index.js:13]: nice work!

File: src/thunks/getQuestions.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import { isLoading, hasErrored, setQuestions } from '../actions'
+
+export const getQuestions = (round, questionCat, difficulty) => {
+  const url = `https://opentdb.com/api.php?amount=5&category=${questionCat}&difficulty=${difficulty}&type=multiple`
```
Comment (lilianaweimer) [inline src/thunks/getQuestions.js:4]: I love that we have a dynamic fetch!

**Pull request merged**